.travis.yml

```diff
@@ -103,21 +113,30 @@ matrix:
             - *linux_deps
 
 before_install:
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+      sudo apt-get update;
+      sudo apt-get install -y python3 python3-venv python3-pip python3-setuptools;      
+    fi
+  - python3 -m pip install --user virtualenv
+  - python3 -m venv buildenv
+  - source buildenv/bin/activate
+  - python3 -m pip install --upgrade pip
+  - pip install scons
+  - scons --version
```

Comment:
This can be simplified a lot by using the pre-existing Python 3 and pip of the Xenial environment. I experimented a bit, I'll make a separate PR to upgrade to Python 3.

Reply:
I pushed the changes as #36074, so if you rebase you should be able to drop all `.travis.yml` changes.

Reply:
thank you!, I will rebase without the edits to those files